src/main/java/Duke.java

```diff
@@ -1,10 +1,178 @@
+                System.out.println("\t Task number provided is not valid. Press \"list\"to see");
+                System.out.println("\t the list of available task numbers");
+                System.out.println(LINE);
+                return;
+            }
+            Task tempTask = l1.get(number);
+            tempTask.done();
+            System.out.println(LINE);
+            System.out.println("\t Nice! I've marked this task as done:");
+            System.out.println("\t   "+tempTask.getTaskDescription());
+            System.out.println(LINE);
+        }else{
+            System.out.println(LINE);
+            System.out.println("\t Please enter the task number to be mark as done!");
+            System.out.println(LINE);
+        }
+    }
+
+    public static void updateTask(ArrayList<Task> l1, String userIn, String timing) {
+        String[] temp=getCommand(userIn);
+        String action="";
+        boolean flip=false;
+        for(int i=1; i<(temp.length); i++) {
+            if (temp[i].charAt(0) == '/') {
+                flip=true;
+            }
+            if(flip){
+                timing += temp[i] + " ";
+            }else{
+                action += temp[i] + " ";
+            }
+        }
+        action=action.trim();
+        timing=timing.trim();
+        timing=timing.replace('/',' ');
+        if(!timing.equals("")){
+            timing+=")";
+        }
+        Task task = new Task(action, temp[0], timing);
+        System.out.println(LINE);
+        l1.add(task);
+        System.out.println("\t Got it. I've added this task:");
+        System.out.println("\t   "+task.getTaskDescription());
+        System.out.println("\t Now you have " + l1.size()+ " tasks in the list.");
+        System.out.println(LINE);
+    }
+
+    public static int validateCommand(String inCommand, ArrayList<Task> l1) {
+        String[] userCommand = getCommand(inCommand);
+        int status;
+        String timing="";
+        switch (userCommand[0]) {
+            case "bye":
+                status = 1;
+                executeBye();
+                break;
+            case "list":
+                status = 2;
+                executeList(l1);
+                break;
+            case "done":
+                status = 3;
+                executeDone(inCommand,l1);
+                break;
+            case "todo":
+                status = 4;
+                updateTask(l1, inCommand, timing);
+                break;
+            case "deadline":
+                status = 5;
+                timing="(by:";
+                updateTask(l1, inCommand, timing);
+                break;
+            case "event":
+                status=6;
+                timing="(at:";
+                updateTask(l1, inCommand, timing);
+                break;
+            case "help":
+                status=7;
+                listCommands();
+                break;
+            default:
+                status = -1;
```

Comment:
Maybe include a break or //Fallthrough in the default case?
This is recorded in the coding standard:
"The explicit //Fallthrough comment should be included whenever there is a case statement without a break statement."